.github/workflows/jeandle-llvm-test.yml

```diff
@@ -10,35 +11,28 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-
-  test:
-    name: "jeandle-llvm test"
-    runs-on: ${{ matrix.runs-on }}
-
-    strategy:
-      matrix:
-        include:
-          - arch: x64
-            target: X86
-            runs-on: ubuntu-22.04
-          - arch: aarch64
-            target: AArch64
-            runs-on: ubuntu-22.04-arm
-
-    steps:
-      - name: 'checkout the source code'
-        uses: actions/checkout@v4
-        with:
-          repository: ${{ github.event.pull_request.head.repo.full_name }}
-          ref: ${{ github.event.pull_request.head.ref }}
-          fetch-depth: 1
-
-      - name: build and test
-        run : |
-          mkdir build
-          cd build
-          cmake -G "Unix Makefiles" -DLLVM_TARGETS_TO_BUILD=${{ matrix.target }} -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_ASSERTIONS=ON -DLLVM_BUILD_LLVM_DYLIB=ON -DLLVM_DYLIB_COMPONENTS=all ../llvm
-          make check-llvm -j $[$(nproc)/2]
+  test-linux-x64:
+    name: linux-x64
+    uses: ./.github/workflows/test-linux.yml
+    with:
+      runs-on: ubuntu-22.04
+      target: X86
+
+  test-linux-aarch64:
+    name: linux-aarch64
+    uses: ./.github/workflows/test-linux.yml
+    with:
+      runs-on: ubuntu-22.04-arm
+      target: AArch64
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'AArch64') }}
+
+  test-linux-riscv64:
+    name: linux-riscv64
+    uses: ./.github/workflows/test-linux.yml
+    with:
+      runs-on: ubuntu-22.04
+      target: RISCV
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'RISC-V') }}
```

Comment:
LLVM can configure more than one target in one compilation, so we can merge separate test jobs in one. Just use labels to determine which targets we need to include.